File: spring-ai-multimodal/src/main/java/com/example/springaimultimodal/AskController.java
package com.example.springaimultimodal;

import org.springframework.ai.chat.client.ChatClient;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.core.io.Resource;
import org.springframework.util.MimeTypeUtils;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class AskController {

  private final ChatClient chatClient;

  @Value("classpath:/forecast.jpg")
  Resource forecastImageResource;

  public AskController(ChatClient.Builder chatClientBuilder) {
    this.chatClient = chatClientBuilder.build();
  }

  @PostMapping("/ask")
  public Answer ask(@RequestBody Question question) throws Exception {
    var answer = chatClient.prompt()
        .user(userSpec -> userSpec
            .text(question.question())
            .media(MimeTypeUtils.IMAGE_JPEG, forecastImageResource))
        .call()
        .content();

    return new Answer(answer);
  }

}

File: spring-ai-kotlin/src/main/kotlin/com/example/demo/AskController.kt
package com.example.demo
import org.springframework.ai.chat.client.ChatClient
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

data class Question(val question: String)
data class Answer(val answer: String)

@RestController
class AskController(chatClientBuilder: ChatClient.Builder) {
    private val chatClient = chatClientBuilder.build()

    @PostMapping("/ask")
    fun ask(@RequestBody question: Question): Answer {
        val answer = chatClient
            .prompt()
            .user(question.question)
            .call()
            .content()
        return Answer(answer)
    }
}
